src/coreclr/vm/jitinterface.cpp

```diff
@@ -10254,6 +10252,31 @@ void CEEInfo::getAsyncInfo(CORINFO_ASYNC_INFO* pAsyncInfoOut)
     EE_TO_JIT_TRANSITION();
 }
 
+CORINFO_CLASS_HANDLE CEEInfo::getContinuationType(
+    size_t dataSize,
+    bool* objRefs,
+    const CORINFO_CONTINUATION_DATA_OFFSETS& dataOffsets)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    CORINFO_CLASS_HANDLE result = NULL;
+
+    JIT_TO_EE_TRANSITION();
+
+    AllocMemTracker amTracker;
+    // TODO: table to share/deduplicate these
+    result = (CORINFO_CLASS_HANDLE)m_pMethodBeingCompiled->GetModule()->CreateContinuationMethodTable((unsigned)dataSize, objRefs, dataOffsets, &amTracker);
```

Comment:
I think these should be created on the loader allocator.

Reply:
That makes a lot of sense. Will change this.

Reply:
The parts of the system that will likely get in trouble are if we (via reflection) get a System.Type of one of these, and the various debugger and profiler apis which examine the heap and various values. Please get @tommcdon or @noahfalk to take a look at this before we signoff.

Reply:
Gotcha. At least that is some evidence that things may not be _badly_ broken with these new types. I suppose they still differentiate themselves in that they have no metadata token available, so that may cause some problems of the "badly broken" kind.

I have opened #120800 about follow-up work here and quoted some of the comments from this comment thread.